tests/serverpod_test_server/test_e2e/http_codes_test.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:convert';
+
+import 'package:http/http.dart' as http;
+import 'package:serverpod_test_server/test_util/config.dart';
+import 'package:test/expect.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('Given an HTTP client invoking a serverpod endpoint method', () {
+    setUpAll(() async {});
+
+    tearDownAll(() async {});
+
+    test(
+        'when calling an endpoint method with correct parameters '
+        'then it should respond with 200 ok', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+          'name': 'Starbase Alpha',
+        }),
+      );
+
+      expect(response.statusCode, 200);
+      expect(response.body, contains('Hello Starbase Alpha'));
+    });
+
+    test(
+        'when calling an endpoint method with missing parameters '
+        'then it should respond with 400 bad request', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+        }),
+      );
+
+      expect(response.statusCode, 400);
+      expect(response.body, contains('Missing required query parameter: name'));
+    });
+
+    test(
+        'when calling an endpoint method with too many parameters '
+        'then it should respond with 400 bad request', () async {
+      var response = await http.post(
+        Uri.parse('${serverUrl}simple'),
+        body: jsonEncode({
+          'method': 'hello',
+          'name': 'Starbase Alpha',
+          'extra': 'spurious value',
+        }),
+      );
+
+      expect(response.statusCode, 400);
+    }, skip: 'desired behavior unclear');
```

Comment:
Suggestion: The desired behavior here is to ignore any additional parameters. We have tests validating this behavior for the parameter parsing.

I think this is usually common practice when parsing query parameters, but it also enables us to be backwards compatible with old clients even if a parameter is removed from an endpoint.

Reply:
Playing devil's advocate:
On the other hand, removing a parameter is a customer-space decision, and perhaps the framework shouldn't force that sort of backwards compatibility and thereby preventing what can be desired error handling.

Also, parameters are sent in the message body so it's not a case of HTTP query convention, rather it is a Serverpod protocol decision.

And finally, getting an intelligable error message when misspelling in Postman or on the Curl command line is superior UX! :)

Reply:
BUT even if so, perhaps now is not the time to change this design which has been decided upon and there are tests for!

Reply:
Good points!

> ...parameters are sent in the message body so it's not a case of HTTP query convention rather it is a Serverpod protocol decision.

Very true, this is a design decision in the Serverpod protocol. We can deviate from normal query parsing conventions here if we have reason to do so.

> ...getting an intelligable error message when misspelling in Postman or on the Curl...

I'm not sure I follow how it would help in this case? 

Regardless, changing this would be a breaking change for existing clients so it is not actionable at this time.